mkt/extensions/models.py

```diff
@@ -35,29 +35,43 @@
 log = commonware.log.getLogger('z.extensions')
 
 
-class ExtensionManager(ManagerBase):
+class ExtensionQuerySet(models.QuerySet):
+    def by_identifier(self, identifier):
+        """Return a single Extension from an identifier, slug or pk."""
+        # Slugs can't contain only digits.
+        if unicode(identifier).isdigit():
+            return self.get(pk=int(identifier))
+        else:
+            return self.get(slug=identifier)
+
     def pending(self):
-        return self.filter(
-            disabled=False, versions__status=STATUS_PENDING)
+        return self.filter(disabled=False, versions__status=STATUS_PENDING)
 
     def public(self):
         return self.filter(disabled=False, status=STATUS_PUBLIC)
```

Comment:
It was a conscious decision not to, but up for debate. I separated the status from the deletion state, so technically, if you want to find all public extensions including deleted, you can with `Extension.objects.public()`, and if you want public non-deleted ones, you'd do `Extension.objects.without_deleted().public()`. This might be over-engineering on my part, let me know what you think.